include/envoy/network/filter.h

```diff
@@ -331,6 +331,13 @@ class FilterChain {
    * const std::vector<FilterFactoryCb>& a list of filters to be used by the new connection.
    */
   virtual const std::vector<FilterFactoryCb>& networkFilterFactories() const PURE;
+
+  /**
+   * @brief Get the Tag object associated with this filter chain.
```

Comment:
What is a tag? It's capitalized, so I assumed an object with its own docs, but it looks like an opaque ID. Since it's just an integer, we need more comments here on what and why.

Reply:
Will add the comment:
The idea of the tag is analogous to the `listener_tag`
* assigned at run time, to identify a filter chain. The tag generator should endeavor to assign the same tag to the new filter chain, if the old filter chain and new one are identical. 
* will be used to identify which `class FilterChain` a connection belongs, so that we can (in follow up PRs) drain all connections when listener decide to destruct a filter chain.

Reply:
ListenerTag here https://github.com/envoyproxy/envoy/blob/587e07974e6badb061ee3c9413660ab47f42750f/include/envoy/network/connection_handler.h#L92